sky/task.py

```diff
@@ -295,10 +296,24 @@ def from_yaml(yaml_path: str) -> 'Task':
 
         task_storage_mounts: Dict[str, storage_lib.Storage] = {}
         all_storages = fm_storages
+        enabled_clouds = global_user_state.get_enabled_clouds()
+        enabled_clouds = [str(cloud) for cloud in enabled_clouds]
+        r2_is_enabled, _ = cloudflare.check_credentials()
+        if r2_is_enabled:
+            enabled_clouds.append('r2')
         for storage in all_storages:
             mount_path = storage[0]
             assert mount_path, \
                 'Storage mount path cannot be empty.'
+            store_type = storage[1]['store']
+            if store_type:
+                cloud_type = storage_lib.STORE_TYPE_TO_CLOUD_TYPE[store_type]
+                if cloud_type not in enabled_clouds:
+                    with ux_utils.print_exception_no_traceback():
+                        raise exceptions.CloudDisabledError(
```

Comment:
Instead of creating a new exception, let's use `exceptions.ResourcesUnavailableError` which is consistent with what we do elsewhere in our code base:

```
(base) ➜  sky-experiments git:(master) ✗ sky launch --cloud aws
sky.exceptions.ResourcesUnavailableError: task_lib.Task Task(run=<empty>)
  resources: AWS() requires AWS which is not enabled. To enable access, run sky check , or change the cloud requirement
```

Reply:
Resolved!